### PR TITLE
Use motion in presmoothing forward/postsmoothing backward projectors

### PR DESCRIPTION
**Pull request by rijobro**
This breaks the current multithreading in STIR (due to changes to distributable.cxx and DistributedWorker.cxx).

Therefore it's currently work in progress 

Can't tag you as an assignee, @ashgillman.

### CODE REVIEW

File: src/recon_buildblock/ForwardProjectorByBin.cxx [hunk 1]
```diff
@@ -187,6 +187,135 @@ forward_project(RelatedViewgrams<float>& viewgrams,
...
+  check(*proj_data.get_proj_data_info_sptr(), *_density_sptr);
+  shared_ptr<DataSymmetriesForViewSegmentNumbers>
+    symmetries_sptr(this->get_symmetries_used()->clone());
+
+  const std::vector<ViewSegmentNumbers> vs_nums_to_process =
+    detail::find_basic_vs_nums_in_subset(*proj_data.get_proj_data_info_ptr(), *symmetries_sptr,
+                                         proj_data.get_min_segment_num(), proj_data.get_max_segment_num(),
+                                         0, 1/*subset_num, num_subsets*/);
+#ifdef STIR_OPENMP
+#pragma omp parallel for  shared(proj_data, _density_sptr, symmetries_sptr) schedule(runtime)
```
Comment (KrisThielemans) [inline src/recon_buildblock/ForwardProjectorByBin.cxx]: delete _density_sptr here, as not used (explictly anyway) so it doesn't compile

File: src/recon_buildblock/ForwardProjectorByBin.cxx [hunk 2]
```diff
@@ -187,6 +187,135 @@ forward_project(RelatedViewgrams<float>& viewgrams,
...
+{
+    actual_forward_project(viewgrams,*_density_sptr,
+                           min_axial_pos_num, max_axial_pos_num,
+                           min_tangential_pos_num, max_tangential_pos_num);
+}
+
+
+void
+ForwardProjectorByBin::
+set_input(const shared_ptr<DiscretisedDensity<3,float> >& density_sptr)
```
Comment (KrisThielemans) [inline src/recon_buildblock/ForwardProjectorByBin.cxx]: same comments as for backprojector

**Comment by KrisThielemans** [comment]
Note: OPENMP is currently broken due to problems with multiple threads backprojecting into the same image (which is now a class-member). Probably will have to take care of the multi-threading inside the backprojector class, i.e. 
1. at `set_up` create a list of images, one for each thread, 
2. at each call, backproject to the image in the relevant thread, 
3. at get_output(), add them all up.

then need to think what to do with the `Postsmoothing*`

**Comment by rijobro** [comment]
This PR has been mostly superceded (except the actual motion components) by #362.

File: src/include/stir/recon_buildblock/BackProjectorByBin.h [hunk 3]
```diff
@@ -112,13 +112,41 @@ class BackProjectorByBin :
...
+void back_project(const RelatedViewgrams<float>&,
+          const int min_axial_pos_num, const int max_axial_pos_num);
+
+ /*! \brief projects the specified range of the viewgrams into the volume
+   it adds to the data already present in the volume.*/
+void back_project(const RelatedViewgrams<float>&,
+          const int min_axial_pos_num, const int max_axial_pos_num,
+          const int min_tangential_pos_num, const int max_tangential_pos_num);
+
+ /*! \brief tell the back projector to start accumulating into a new image*/
```
[the remark this answers is not in the document]
Comment (rijobro) [reply src/include/stir/recon_buildblock/BackProjectorByBin.h]: done in #239

File: src/experimental/motion_test/CMakeLists.txt [hunk 4]
```diff
@@ -23,17 +23,23 @@ set(dir_INVOLVED_TEST_EXE_SOURCES ${dir}_INVOLVED_TEST_EXE_SOURCES)
...
 			${CMAKE_CURRENT_BINARY_DIR}/test_BSpline_transformations 
 				${CMAKE_CURRENT_SOURCE_DIR}/input/to_transform.nii.gz
 				${CMAKE_CURRENT_SOURCE_DIR}/input/ground_truth.nii.gz
 				${CMAKE_SOURCE_DIR}/examples/nifti/disp_4D.nii.gz)
+
+		ADD_TEST(test_ProjectorsWithMotion 
+			${CMAKE_CURRENT_BINARY_DIR}/test_ProjectorsWithMotion
+				/Users/rich/Documents/Data/Kjell-patlak/Output/09_Movement/1_LR/dyn_im_LR_4.nii
+				/Users/rich/Documents/Data/Kjell-patlak/Output/13_Prompts/1_LR/prompts_LR_4.hs
+				/Users/rich/Documents/Data/Kjell-patlak/Output/10_DispFields/disp_LR.nii
```
Comment (KrisThielemans) [inline src/experimental/motion_test/CMakeLists.txt]: obviously cannot have this

**Comment by KrisThielemans** [comment]
how useful is this PR actually? The test doesn't really seem to do anything? If we keep it, can you please change the title? Or to reduce confusion, just abandon this and create a new PR, as 99% of the comments are obsolete.

There's one Travis job that failed weirdly with
```
/Users/travis/build/UCL/STIR/src/experimental/motion_test/test_ProjectorsWithMotion.cxx:168:21: error: no member named 'start_accumulating_in_new_image' in 'stir::PostsmoothingBackProjectorByBin'
    back_projector->start_accumulating_in_new_image();
    ~~~~~~~~~~~~~~  ^
```

**Comment by rijobro** [comment]
doesn't seem to do much and the pre-smoothing-fwd and post-smoothing-back were superseded by https://github.com/UCL/STIR/pull/367.

**Pull request closed without merging**